com.microsoft.mrtk.windowsspeech/Subsystems/TextToSpeech/WindowsTextToSpeechSubsystem.cs

```diff
@@ -0,0 +1,175 @@
+        static void Register()
+        {
+            // Fetch subsystem metadata from the attribute.
+            var cinfo = XRSubsystemHelpers.ConstructCinfo<WindowsTextToSpeechSubsystem, TextToSpeechSubsystemCinfo>();
+
+            if (!WindowsTextToSpeechSubsystem.Register(cinfo))
+            {
+                Debug.LogError($"Failed to register the {cinfo.Name} subsystem.");
+            }
+        }
+
+        [Preserve]
+        class WindowsTextToSpeechSubsystemProvider : Provider
+        {
+            public WindowsTextToSpeechSubsystemProvider() : base()
+            { }
+
+            public override void Destroy()
+            {
+#if WINDOWS_UWP
+                if (synthesizer != null)
+                {
+                    synthesizer.Dispose();
+                    synthesizer = null;
+                }
+#endif // WINDOWS_UWP
+            }
+
+            #region ITextToSpeechSubsystem implementation
+
+#if !(WINDOWS_UWP || UNITY_STANDALONE_WIN || UNITY_EDITOR_WIN)
+            private bool haveLogged = false;
+#endif
+
+            /// <inheritdoc/>
+            public override async Task<bool> TrySpeak(string phrase, AudioSource audioSource)
+            {
+                if (audioSource == null)
+                {
+                    Debug.LogError("Must specify the AudioSource object on which the text to speech data should be applied.");
+                    return false;
+                }
+
+                byte[] waveData = await Synthesize(phrase);
+                if (waveData == null)
+                {
+                    return false;
+                }
+
+                // Convert from bytes to floats and create an AudioClip.
+                if (!TextToSpeechHelpers.TryConvertWaveData(
+                        waveData,
+                        out int samples,
+                        out int sampleRate,
+                        out int channels,
+                        out float[] audioFloats))
+                {
+                    Debug.LogError("Failed to convert speech audio format.");
+                    return false;
+                }
+
+                audioSource.clip = TextToSpeechHelpers.CreateAudioClip(
+                    "SynthesizedText",
+                    audioFloats,
+                    samples,
+                    channels,
+                    sampleRate);
+
+                audioSource.Play();
+                
+                return true;
+            }
+
+            /// <summary>
+            /// Synthesizes the specified phrase.
+            /// </summary>
+            /// <param name="phrase">The phrase to be synthesized.</param>
+            /// <returns>The audio (wave) data upon successful synthesis, or null.</returns>
+            private async Task<byte[]> Synthesize(string phrase)
+            {
+                if (string.IsNullOrWhiteSpace(phrase))
+                {
+                    Debug.LogWarning("Nothing to speek");
```

Comment:
```suggestion
                    Debug.LogWarning("Nothing to speak");
```

Reply:
haha... the perils of tyoing too fast :smile: